server.go

```diff
@@ -243,6 +245,36 @@ func (s *Server) register(sd *ServiceDesc, ss interface{}) {
 	s.m[sd.ServiceName] = srv
 }
 
+// ServiceMetadata returns the metadata for a service or method.
+// service should be the full service name with package, in the form of <package>.<service>.
+// method should be the method name only.
+// If only service is important, method should be an empty string.
+func (s *Server) ServiceMetadata(service, method string) interface{} {
```

Comment:
@iamqizhao  The reason for not returning an error is that even if the returned error is nil, caller still needs to check whether the returned metadata is nil.

@puellanivis  The returned value is needed by reflection. So only checking if the service is valid is not sufficient.
Server reflection is implemented in a different package. So even if this function is only used by server reflection, it still needs to be exported. The problem is this function becomes confusing for users other than reflection.
We will try to come up with another way to accomplish this.